pkg/trace/stats/aggregation.go

```diff
@@ -44,12 +47,13 @@ type BucketsAggregationKey struct {
 
 // PayloadAggregationKey specifies the key by which a payload is aggregated.
 type PayloadAggregationKey struct {
-	Env          string
-	Hostname     string
-	Version      string
-	ContainerID  string
-	GitCommitSha string
-	ImageTag     string
+	Env           string
+	Hostname      string
+	Version       string
+	ContainerID   string
+	GitCommitSha  string
+	ImageTag      string
+	ContainerTags string
```

Comment:
How many tags do we expect to have in `ContainerTags`? The existing design assumes a 1:1 relationship between ContainerID and ContainerTags, and as such, putting the tags in the aggregation key is unnecessary.

Reply:
It is an 1:1 mapping. We need this field because the processing and flush are async. We need to retain the `ContainerTags` either in a cache (as in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31642) or keep it here, while cache is even worse.

Reply:
I would prefer for us to do this the way we did it for `peerTags`:
1. Keep a hash in the key struct itself.
2. The key struct will point to an aggregation struct that has the actual slice of container tag strings.

Reply:
Looking at this further I think a different approach might work? See my comment here: https://github.com/DataDog/datadog-agent/pull/23503/files#r1581607170